small-talk/app/login/page.js

```diff
@@ -28,6 +29,14 @@ export default function LoginPage() {
       alert("All fields are necessary!");
       return ok;
     }
+    if(!(await validateInput(username))){
+      alert("INVALID"); // used less descriptive error code to hide the checks we use
+      return;
+    }
+    if(!(await validateInput(password))){
+      alert("INVALID"); // used less descriptive error code to hide the checks we use
+      return;
+    }
```

Comment:
Could combine these to one check if the alert is the same

Reply:
Thank you for this suggestion also. I have implemented and re-pushed